### PR TITLE
add app index

### PR DESCRIPTION
**Pull request by drowsy810301**
add app index

### CODE REVIEW

File: index/templates/newindex.html [hunk 1]
```diff
@@ -0,0 +1,109 @@
+<!--
```
Comment (drowsy810301) [inline index/templates/newindex.html]: Sorry that I forgot to change the filename,and I just changed the filename to index.html

File: index/static/js/index.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+var displayinfo=false;
+var logininfo=false;
```
Comment (drowsy810301) [inline index/static/js/index.js]: OK,It's done.

File: index/static/js/index.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+window.onload = function() {
+    alert("You have new messages!");
+}
+function info() {
+    if(displayinfo==false){
```
Comment (drowsy810301) [inline index/static/js/index.js]: OK,It's done.

File: index/static/js/index.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+var displayinfo=false;
+var logininfo=false;
+
+window.onload = function() {
+    alert("You have new messages!");
+}
+function info() {
+    if(displayinfo==false){	
+        document.getElementById("information").style.display="block";
+        displayinfo=!displayinfo;
```
Comment (drowsy810301) [inline index/static/js/index.js]: OK,It's done.

File: index/static/js/index.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+}
+function sendinfo() {
+    var text=document.getElementById("txt").value;
+    var orgintext=document.getElementById("information").innerHTML;
+    document.getElementById("information").innerHTML=orgintext+'<li class="divider">'+text+'</li>'
+    //alert(orgintext);
+    alert("You have new messages!"+text);
+}
+function log_in() {
+    //alert('1');
```
Comment (drowsy810301) [inline index/static/js/index.js]: OK,It's done.

File: index/templates/base.html [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+        </div>
+        <div class="form-group">
+          <label class="control-label" for="inputSmall">Password</label>
+          <input class="txtbd form-control input-sm" type="text" id="inputSmall " placeholder="Password">
+          <a href="#forget-popup" class="open-popup-link"><p class="p6">forget password?</p></a>
+        </div>
+
+        <div class="modal-footer">
+          <br>
+          <div class="poptxt_login">Don't have an account? You can <a href="#register-popup" class="open-popup-link"><p class="p6">register here</p></a></div>
```
Comment (drowsy810301) [inline index/templates/base.html]: I rearranged the code.Is it okay now?

File: index/templates/base.html [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+    <input type="text" id='txt' style="width:150px";>
+    <input type="button" style="cursor: pointer;" value="send" onclick="sendinfo();">
+  </div>
+
+  {% block body_block %}
+  {% endblock body_block %}
+  <!-- footer -->
+  <div class="footer fooe" style="height: 40px">
+    <a href="" class="p11"  style="position:relative;left:30px;">Contact us</a>       
+    <p class="p10" style="position:relative;float: right;right:15px;top:5px">© NTHU Online Judge 2013, Templates are from Bootstrap Project</p>
```
Comment (drowsy810301) [inline index/templates/base.html]: I chaged 2013 to 2014.

File: nthuoj/urls.py [hunk 8]
```diff
@@ -3,4 +3,12 @@
 
 urlpatterns = patterns('',
     url(r'^admin/', include(admin.site.urls)),
+    url(r'^$', 'index.views.home'),
+    url(r'^broken/', 'index.views.broken'),
+    url(r'^base/', 'index.views.base'),
+    url(r'^get_time/', 'index.views.get_time'),
```
Comment (drowsy810301) [inline nthuoj/urls.py]: I don't know what you mean...
do you mean the url in line6,7,8,9 is useless?

File: index/static/css/index.css [hunk 9]
```diff
@@ -0,0 +1,298 @@
...
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+body{
+  padding-top:0px;
```
Comment (nickelcheng) [inline index/static/css/index.css]: Add a space in front of 0px?

File: index/templates/base.html [hunk 10]
```diff
@@ -0,0 +1,263 @@
...
+          <ul class="dropdown-menu">
+            <li><a href="#">contest archive</a></li>
+            <li><a href="#">running contest</a></li>
+          </ul>
+        </li>
+        <li class="dropdown">
+          <a href="#" class="dropdown-toggle" data-toggle="dropdown">
+            <p class="p2">Problem<b class="caret"></b></p>
+          </a>
+          <ul class="dropdown-menu">
```
Comment (nickelcheng) [inline index/templates/base.html]: Please don't hard code volumes. Get number of problems and use a for loop to generate them.

File: index/templates/clock.html [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+-->
+  <p id='time' style="font-size: 18px;color:black">{{ tstr }}</p>
```
Comment (nickelcheng) [inline index/templates/clock.html]: There is still no newline OAQ?

File: index/templates/index.html [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+    </thead>
+    <thead style="background-color: #E5E5E5">
+      <tr>
+        <th>#</th>
+        <th>Contest name</th>
+        <th>remaining time</th>
+        <th>upcoming(1 day)</th>
+      </tr>
+    </thead>
+    <tbody class="line" >
```
Comment (nickelcheng) [inline index/templates/index.html]: You should not limit the columns to 6. Use for loop instead.

File: index/templates/base.html (hunk 10 above)
Comment (drowsy810301) [inline index/templates/base.html]: Sorry.I know.I haven't changed yet.If I changed,I would write in the commit comment.

**Comment by bruce3557** [comment]
I think this PR can be merged if @nickelcheng has no other suggestion.
Thanks @drowsy810301 !!

**Comment by bruce3557** [comment]
BTW, @drowsy810301  you need to merge latest code and put it here.
Than I can merge this.

**Comment by nickelcheng** [comment]
There is still some hard code need to be fixed.

**Comment by nickelcheng** [comment]
I'm ok with this PR.

File: index/templates/base.html [hunk 13]
```diff
@@ -0,0 +1,270 @@
...
+            <label class="control-label" for="inputSmall">
+              Password
+            </label>
+            <input class="form-control input-sm" type="text" 
+                          id="inputSmall" placeholder="Password">
+            <label class="control-label" for="inputSmall">
+              Reenter Password
+            </label>
+            <input class="form-control input-sm" type="text" 
+                          id="inputSmall" placeholder="Reenter Password">
```
Comment (nickelcheng) [inline index/templates/base.html]: I suggest "id" align with "class"

File: index/templates/base.html [hunk 14]
```diff
@@ -0,0 +1,270 @@
...
+            <label class="control-label" for="inputSmall">
+              Email
+            </label>
+            <input class="form-control input-sm" type="text" 
+                          id="inputSmall" placeholder="Email">
```
Comment (nickelcheng) [inline index/templates/base.html]: I suggest "id" align with "class"

File: index/views.py [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+                            'info1':123, 'info2':1234567})
+
+def get_time(request):
+    t = time.time()
+    tstr = datetime.datetime.fromtimestamp(t).strftime('%Y-%m-%d %H:%M:%S')
+    return HttpResponse(tstr)
+
+def submit(request):
+    return render(request, 'submit.html', 
+                            {'people':'123', 'info1':123, 'info2':1234567})
```
Comment (bruce3557) [inline index/views.py]: Is this to test or another purpose?
not sure the idea about ```'people': 123```
Could you please explain it?

File: nthuoj/settings.py [hunk 16]
```diff
@@ -36,7 +36,11 @@
     'django.contrib.sessions',
     'django.contrib.messages',
     'django.contrib.staticfiles',
+<<<<<<< HEAD
+    'index',
+=======
     'users',
+>>>>>>> 654949c4f754f18ef1967c591d509451e6a99545
```
Comment (bruce3557) [inline nthuoj/settings.py]: Please remove the git message and check what is correct code

File: nthuoj/urls.py [hunk 17]
```diff
@@ -3,6 +3,16 @@
...
+    url(r'^broken/', 'index.views.broken'),
+    url(r'^get_time/', 'index.views.get_time'),
+    url(r'^status/', 'index.views.status'),
+    url(r'^submit/', 'index.views.submit'),
+    url(r'^group_list/', 'index.views.group_list'),
+    url(r'^team_list/', 'index.views.team_list'),
+=======
     url(r'^submit/$', 'users.views.submit'),
     url(r'^profile/$', 'users.views.profile'),
+>>>>>>> 654949c4f754f18ef1967c591d509451e6a99545
```
Comment (bruce3557) [inline nthuoj/urls.py]: Ditto

File: index/views.py (hunk 15 above)
Comment (bruce3557) [inline index/views.py]: Temporarily OK with these fake data.
Please remove it after database model is done.
Thanks

Comment (drowsy810301) [inline index/views.py]: It means the number of online people.I changed it to the random number.

File: nthuoj/settings.py (hunk 16 above)
Comment (drowsy810301) [inline nthuoj/settings.py]: OK,It's done.

File: nthuoj/urls.py (hunk 17 above)
Comment (drowsy810301) [inline nthuoj/urls.py]: OK,It's done.

**Comment by nickelcheng** [comment]
Going to merge. Thank @drowsy810301's effort.

**Pull request merged**